In [1]:
import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler

In [2]:
# Import the data
lending_df = pd.read_csv("Resources/lending_data.csv")
lending_df.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  loan_status  
0                 1       22800            0  
1                 0       13600            0  
2                 0       16100            0  
3                 1       22700            0  
4                 1       23000            0

In [3]:
lending_df['loan_status'].unique()

array([0, 1], dtype=int64)

In [4]:
lending_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77536 entries, 0 to 77535
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   loan_size         77536 non-null  float64
 1   interest_rate     77536 non-null  float64
 2   borrower_income   77536 non-null  int64  
 3   debt_to_income    77536 non-null  float64
 4   num_of_accounts   77536 non-null  int64  
 5   derogatory_marks  77536 non-null  int64  
 6   total_debt        77536 non-null  int64  
 7   loan_status       77536 non-null  int64  
dtypes: float64(3), int64(5)
memory usage: 4.7 MB


In [5]:
# Split the data into X_train, X_test, y_train, y_test
X = lending_df.drop(columns=['loan_status'])
y = lending_df['loan_status']

In [6]:
X_train, X_test, y_train, y_test = train_test_split(X,y)

In [7]:
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
X.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  
0                 1       22800  
1                 0       13600  
2                 0       16100  
3                 1       22700  
4                 1       23000

In [8]:
#scaling data on the values
scaler = StandardScaler()
scaler.fit(X_train)
X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

# predicitons 

before the test are done i belive that the Logistic Regression model will be faster and produce a better score because the structore of this test is to see if a loan will fall in to one or two catigoties approved or not.
where the Random Forest Classifier model is better used for multiple decision trees

In [9]:
# Train a Logistic Regression model print the model score
# In regression analysis, logistic regression (or logit regression) is estimating the parameters of a logistic model (the coefficients in the linear combination).
# got the definition from google 
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression().fit(X_train_scaled, y_train)

In [10]:
print(f"Training Data Score: {classifier.score(X_train_scaled, y_train)}")
print(f"Testing Data Score: {classifier.score(X_test_scaled, y_test)}")

Training Data Score: 0.9941188609162196
Testing Data Score: 0.9941704498555509


In [11]:
# Train a Random Forest Classifier model and print the model score
# The random forest is a supervised learning algorithm that randomly creates and merges multiple decision trees into one “forest.”
# got the definition from google 
from sklearn.ensemble import RandomForestClassifier
clf = RandomForestClassifier(n_estimators=500).fit(X_train_scaled, y_train)

In [12]:
print(f'Training Score: {clf.score(X_train_scaled, y_train)}')
print(f'Testing Score: {clf.score(X_test_scaled, y_test)}')

Training Score: 0.9971970009629936
Testing Score: 0.9920037144036319


# Outcomes 

both the modles worked very well, both scoring in the uptter 99's with Logistic Reggression Model (LRM) scoring (99.4%) and Random Forest Classifier model (RFCM) scoring (99.2%) with a diffrence of only 0.2% favoring the (LRM). but the major diffrence is the amount of time the (RFCM) took to complete comepared to the (LRM), with the  (LRM) finishing in under a second the (RFCM) took over 13.1 seconds, with speed favoring the (LRM) in both catigoties score and spped the (LRM) is the better choice out of the to

In [13]:
# 'worked with TA's and Other students to develop code and also looked some stuff up onon stackoverflow to help me find the answers need to get the right idea going'